src/components/CIPipelineN/AdvancedConfigOptions.tsx

```diff
@@ -179,6 +181,51 @@ export default function AdvancedConfigOptions({
         }
     }
 
+    const targetPlatformMap = getTargetPlatformMap()
+    const [selectedTargetPlatforms, setSelectedTargetPlatforms] = useState<OptionType[]>([])
+    const [showCustomPlatformWarning, setShowCustomPlatformWarning] = useState<boolean>(false)
+
+    const updateSelectedPlatformsCustomTargetPlatform = () => {
+        let _customTargetPlatform = false
+        let _selectedPlatforms = []
+        let targetPlatform = ''
+        if (parentState.selectedCIPipeline?.dockerConfigOverride?.ciBuildConfig?.dockerBuildConfig?.targetPlatform) {
+            targetPlatform =
+                parentState.selectedCIPipeline.dockerConfigOverride.ciBuildConfig.dockerBuildConfig.targetPlatform
+        } else if (parentState.ciConfig?.ciBuildConfig) {
+            if (parentState.ciConfig.ciBuildConfig.dockerBuildConfig?.targetPlatform) {
+                targetPlatform = parentState.ciConfig.ciBuildConfig.dockerBuildConfig.targetPlatform
+            }
+        }
+        _selectedPlatforms = targetPlatform.split(',').map((platformValue) => {
+            _customTargetPlatform = _customTargetPlatform || !targetPlatformMap.get(platformValue)
+            return { label: platformValue, value: platformValue }
+        })
+        setSelectedTargetPlatforms(_selectedPlatforms)
+        setShowCustomPlatformWarning(_customTargetPlatform)
+    }
+
+    const handleChangeInTargetPlatforms = () => {
+        const _form = { ...formData }
+        let platformsArray = []
+        selectedTargetPlatforms.forEach(function (o) {
+            platformsArray.push(o.label)
+        })
+        if (_form.dockerConfigOverride?.ciBuildConfig?.dockerBuildConfig) {
+            _form.dockerConfigOverride.ciBuildConfig.dockerBuildConfig.targetPlatform = platformsArray.join()
+        }
+        setFormData(_form)
+    }
+
+    useEffect(() => {
+        updateSelectedPlatformsCustomTargetPlatform()
```

Comment:
we should add a failsafe empty check on parentState

Reply:
added the failsafe check